Source/Core/Core/IOS/DI/DI.cpp

```diff
@@ -100,16 +100,18 @@ IPCCommandResult DI::IOCtlV(const IOCtlVRequest& request)
     _dbg_assert_msg_(IOS_DI, request.in_vectors[2].address == 0,
                      "DVDLowOpenPartition with cert chain");
 
-    u64 const partition_offset = ((u64)Memory::Read_U32(request.in_vectors[0].address + 4) << 2);
-    DVDInterface::ChangePartition(partition_offset);
+    const u64 partition_offset =
+        static_cast<u64>(Memory::Read_U32(request.in_vectors[0].address + 4)) << 2;
+    const DiscIO::Partition partition(partition_offset);
+    DVDInterface::ChangePartition(partition);
 
     INFO_LOG(IOS_DI, "DVDLowOpenPartition: partition_offset 0x%016" PRIx64, partition_offset);
 
     // Read TMD to the buffer
-    const IOS::ES::TMDReader tmd = DVDInterface::GetVolume().GetTMD();
+    const IOS::ES::TMDReader tmd = DVDInterface::GetVolume().GetTMD(partition);
```

Comment:
For consistency with the other places, can you store the result of `DVDInterface::GetVolume()` in a local and use that instead?
Or, if it doesn't matter in practise, feel free to ignore all related comments (since `GetVolume` doesn't do much except returning a static variable)